examples/demo_runner.py

```diff
@@ -237,16 +240,47 @@ def init_common(self):
 
         return start_state
 
-    def benchmark(self, settings):
-        self.set_sim_settings(settings)
+    def _bench_target(self, _idx=0):
         self.init_common()
 
-        perf = self.do_time_steps()
+        best_perf = None
+        for _ in range(3):
+
+            if _barrier is not None:
+                _barrier.wait()
+                if _idx == 0:
+                    _barrier.reset()
+
+            perf = self.do_time_steps()
+            if best_perf is None or perf["fps"] > best_perf["fps"]:
```

Comment:
This should probably have a comment on why taking the max is a common/legitimate thing in benchmarking.  Since I did this, here is my reasoning:

```
The variance introduced between runs is due to the worker threads being interrupted a different number of times by the kernel, not due to difference in the speed of the code itself.  The most accurate representation of the performance would be a run where the kernel never interrupted the workers, but this isn't feasible,  so we just take the run with the least number of interrupts (the fastest) instead.
```